libmamba/src/solv-cpp/pool.cpp

```diff
@@ -0,0 +1,255 @@
+        -> DependencyId
+    {
+        // Note: libsolv cannot report failure to allocate
+        ::Id const id = ::pool_rel2id(
+            raw(),
+            name_id,
+            version_id,
+            flag,
+            /* .create= */ 1
+        );
+        assert(id != 0);
+        assert(ISRELDEP(id));
+        return id;
+    }
+
+    auto ObjPool::get_dependency_name(DependencyId id) const -> std::string_view
+    {
+        assert(ISRELDEP(id));
+        return ::pool_id2str(raw(), id);
+    }
+
+    auto ObjPool::get_dependency_version(DependencyId id) const -> std::string_view
+    {
+        assert(ISRELDEP(id));
+        return ::pool_id2evr(raw(), id);
+    }
+
+    auto ObjPool::get_dependency_relation(DependencyId id) const -> std::string_view
+    {
+        assert(ISRELDEP(id));
+        return ::pool_id2rel(raw(), id);
+    }
+
+    auto ObjPool::dependency_to_string(DependencyId id) const -> std::string
+    {
+        assert(ISRELDEP(id));
+        return ::pool_dep2str(
+            // Not const in because function may allocate in pool tmp alloc space
+            const_cast<::Pool*>(raw()),
+            id
+        );
+    }
+
+    auto ObjPool::select_solvables(const ObjQueue& job) const -> ObjQueue
+    {
+        ObjQueue solvables = {};
+        ::selection_solvables(
+            // Not const in because function may allocate in pool tmp alloc space
+            const_cast<::Pool*>(raw()),
+            // Queue selection param is not modified
+            const_cast<::Queue*>(job.raw()),
+            solvables.raw()
+        );
+        return solvables;
+    }
+
+    void ObjPool::create_whatprovides()
+    {
+        ::pool_createwhatprovides(raw());
+    }
+
+    auto ObjPool::add_repo(std::string_view name) -> std::pair<RepoId, ObjRepoView>
+    {
+        auto* repo_ptr = ::repo_create(
+            raw(),
+            ::pool_id2str(raw(), add_string(name))  // Shenanigan to make it string_view compatible
+        );
+        // No function exists to create a repo id
+        assert(raw()->repos[raw()->nrepos - 1] == repo_ptr);
+        return { raw()->nrepos - 1, ObjRepoView{ repo_ptr } };
+    }
+
+    auto ObjPool::has_repo(RepoId id) const -> bool
+    {
+        return (id > 0) && (id < raw()->nrepos) && (raw()->repos[id] != nullptr);
+    }
+
+    auto ObjPool::get_repo(RepoId id) -> std::optional<ObjRepoView>
+    {
+        if (!has_repo(id))
+        {
+            return std::nullopt;
+        }
+        return { ObjRepoView{ ::pool_id2repo(raw(), id) } };
+    }
+
+    auto ObjPool::get_repo(RepoId id) const -> std::optional<ObjRepoViewConst>
+    {
+        if (!has_repo(id))
+        {
+            return std::nullopt;
+        }
+        // Safe because we make the Repo deep const
+        return { ObjRepoViewConst{ ::pool_id2repo(const_cast<::Pool*>(raw()), id) } };
+    }
+
+    auto ObjPool::n_repos() const -> std::size_t
```

Comment:
Consider renaming this to `repo_count()` or `count_repos()` if it's usage is strictly for our C++ codebase.